Fix package

syohex opened the pull request:
- Fix `setq-local`(insert space between function and symbol) and use `setq` instead of `setq-local` because `font-lock-defaults` is declared as buffer local variable. (And `setq-local` requires Emacs 24.3 or higher versions)
- Use defvar for global variable declaration
- Add autoload cookie for lazy loading
- Add meta comment

syohex committed:
Insert space between function and symbol

And use setq instead of setq-local because font-lock-defaults
is declared as buffer local variable.

syohex committed:
Use defvar for declaring global variable

syohex committed:
Add autoload cookie for lazy loading

syohex committed:
Add Code: meta comment

Yuki-Inoue commented on cql-mode.el:
```diff
@@ -20,35 +20,38 @@
+              (sql-product-font-lock-syntax-alist)))
 
   )
 
```
add-to-list にも、 autoload をつけるものなんですね。。。？

Yuki-Inoue commented on cql-mode.el:
```diff
@@ -20,35 +20,38 @@
   )
 
+;;;###autoload
 (add-to-list 'auto-mode-alist '("\\.cql\\'" . cql-mode))
```
add-to-list にも、 autoload はつけるものなんですね。。？

Yuki-Inoue committed:
Fix the cql-mode-font-lock-keywords definition

The sql-mode-font-lock-object-name is to be included in the definition
of cql-mode-font-lock-keywords.  It highlights the table name.  For
example,

    create table hoge()
                 ^^^^

Making the sql-mode-font-lock-object-name included in the cql font lock
makes above syntax highlighting available.

Yuki-Inoue commented:
@syohex I noticed `setq`ing the `sql-mode-font-lock-object-name` was not the right refactor. I've sent you the fix which would make it work as I intended.

I'm planning to merge this PR after my PR to your `fix-project` branch is merged.
(I'm not so much familiar with github PR workflow, so point me out if I'm doing something odd.)

syohex commented:
Oh, sorry I mistook by tab indentation.

syohex commented:
I have cherry-picked your commit.
Sorry my mistaking and thanks for patch.
